test/hotspot/gtest/runtime/test_os_windows.cpp

```diff
@@ -703,6 +703,101 @@ TEST_VM(os_windows, reserve_memory_special) {
   TestReserveMemorySpecial_test();
 }
 
+TEST_VM(os_windows, large_page_init_multiple_sizes) {
+    // Enable large pages and set conditions for the test
+    UseLargePages = true;
+    bool schedules_all_processor_groups = true; // Simulate OS version 11 or higher
+    EnableAllLargePageSizes = true; // Assume this flag enables support for all large page sizes
+
+    // Determine the minimum page size
+    const size_t min_size = os::GetLargePageMinimum();
+
+    // Scenario 1: Set LargePageSizeInBytes to 4 times the minimum page size
+    LargePageSizeInBytes = 4 * min_size; // Set a value for multiple page sizes
+
+    // Initialize large page settings for the first scenario
+    os::large_page_init();
+
+    // Verify that large pages are enabled
+    EXPECT_TRUE(UseLargePages) << "UseLargePages should be true after initialization for 4*min_size";
+
+    // Verify that _large_page_size is greater than the default page size
+    const size_t default_page_size = os::vm_page_size();
+    EXPECT_GT(os::_large_page_size, default_page_size) << "Large page size should be greater than the default page size for 4*min_size";
```

Comment:
Assuming I understand the comment correctly, I think it should read "Large page size of 4*min_size should be greater than the default page size"